test/functions.coffee

```diff
@@ -572,3 +572,17 @@ test "#4657: destructured array parameters", ->
   result = f [1, 2, 3, 4]
   arrayEq result.a, [1, 2, 3]
   eq result.b, 4
+
+test "#5013: return statements in parentheses", ->
+  compile = (code) -> CoffeeScript.compile(code, bare: yes).trim().replace(/\s+/g, " ")
+
+  eq "(function() {});",         compile("-> (return)")
+  eq "(function*() {});",        compile("-> (yield return)")
+  eq "(async function() {});",   compile("-> (await return)")
```

Comment:
Why do we test these against generated JS?

Why is there no `return` in the JS?

Reply:
`yield return`, `await return` are ways to mark a function as a generator/async without needing to use `yield` or `await` in the body.

Reply:
Yes but it seems incorrect for there to be `yield`, `await` and `return` in the CoffeeScript and for none of those to appear in the generated JavaScript. They’re not CS-only keywords, they’re basically passthroughs. When else do we swallow a `return` in the CoffeeScript and not output it in the JavaScript?

I would think that `-> (yield return)` should generate:

```js
(function*() {
  (yield return);
});
```

which is invalid JavaScript; but that’s okay. It’s what the user typed.

Reply:
We have [been slicing `Return`s off the end](https://github.com/jashkenas/coffeescript/blob/master/src/nodes.coffee#L484) of `Block`s since the very beginning(™).